main.go

```diff
@@ -3,30 +3,80 @@ package main
 import (
 	"log"
 	"net"
+	"strings"
 
 	"git.icyphox.sh/taigobot/db"
 	"git.icyphox.sh/taigobot/plugins"
 	"github.com/dgraph-io/badger/v3"
 	"gopkg.in/irc.v3"
 )
 
+// (nojusr): A simple check func to find out if an irc message
+// is supposed to be a CTCP message
+func isCTCPmessage(m *irc.Message) bool {
+	// NOTE(nojusr): CTCP messages are identified by their first character byte
+	// being equal to 0x01.
+	if m.Params[1][0] == 0x01 {
+		return true
+	}
+	return false
+}
+
+func handleCTCPMessage(c *irc.Client, m *irc.Message) {
+
+	// (nojusr): Refer to for further commands to implement and as a general
+	// guide on how CTCP works:
+	// https://tools.ietf.org/id/draft-oakley-irc-ctcp-01.html
+
+	var ctcpCommand = m.Params[1]
+
+	// NOTE(nojusr): start of the main if/else tree for CTCP checking. idk why, but
+	// for some reason a straight switch/case comparison simply did not work
+	if strings.Contains(ctcpCommand, "VERSION") {
+
+		c.WriteMessage(&irc.Message{
+			Command: "PRIVMSG",
+			Params: []string{
+				m.Prefix.Name,
+				// TODO(nojusr): read version from config
+				string(0x01) + "VERSION NONE OF YOUR BUISSNESS" + string(0x01),
```

Comment:
```suggestion
				fmt.Sprintf("0x01%s0x01", version)
```
where version info is not some meme shit, and is loaded from config.